src/csc439teamllama/cardgame/GolfGameModel.java

```diff
@@ -56,13 +56,80 @@ public int PlayerIndex(){
      * and prints it to the screen.
      */
     protected void ScoreboardUpdate(){
+
+        //Update player scores
+        UpdateScores();
+
+        //Set the scoreBoard array to be equal to the players array
         for(int i = 0; i < players.length; i++){
             scoreBoard[i] = players[i];
         }
 
+        //Sort the scoreBoard array
         Arrays.sort(scoreBoard);
 
-        //This uses the displayScoreBoard method that Tedla created. It is commented out at the moment.
+        //This uses the displayScoreBoard method to display the updated scores of the players
         //view.displayScoreBoard(this);
     }
+
+    protected void UpdateScores(){
+
+        for(int i = 0; i < players.length; i++){
+            players[i].score = 0;
+            for(int j = 0; j < 6; j++) {
+
+                /*
+                These lines compensate for columns that are equal. If the two cards in a column are equal, then their scores are skipped.
+                These conditionals all start by checking that the cards in question are face up, then they check to see that the current card is
+                equal to the one under it. If this is the case, then the loop continues on to the next iteration.
+                */
+                if (players[i].hand[0].getFacing() == playingCard.Facing.UP && players[i].hand[3].getFacing() == playingCard.Facing.UP &&
```

Comment:
Can be improved to use a ma thematic approach to compare cols, for 0-2 of loop we would check j against card j+3, and then once we get to 3 we would compare the earlier 3 cards with either a modulo, which means we can condense it down to one line instead of 6, or simple subtract 3 from j,. condensing from 6 to 2.